### ISSUE TITLE
content selector - need select only top child elements

### ISSUE THREAD

**Issue by shuslav** [yyx990803/vue#737]
If I place my component inside another same component, e.g.:
```html
    <my-comp>
        <div class="body">
            <my-comp>
                <div class="body">

                </div>
            </my-comp>
        </div>
    </my-comp>
```
The content selector of the outer component selects two DIVs (its own and nested component's):
```html
    <content select="div.body"></content>
```
Is it possible to select 'div.body' of outer component only?

**Comment by noirbizarre** [comment]
I have exactly the same problem.

I have some component designed to use the top child ``<footer>`` tag as a container and I had some bad surprise when I started nesting them.

Writing the following select all child element, whatever deep they are.
```html
<content select="footer"></content>
```

The ``>`` css selector is not working on root and I can't write:
```html
<content select="> footer"></content>
```
But I think this can be a good way to set the restriction, because maybe some people need to select all children.

**Comment by shuslav** [comment]
I've found workaround for my issue:

```html
    <my-comp>
        <div>          <!-- add root element -->
          <div class="body">
              <my-comp>
                 <div>  <!-- add root element -->
                    <div class="body">

                    </div>
                  </div>
              </my-comp>
          </div>
        </div>
    </my-comp>
```
The selector:
```html
    <content select=":only-child > div.body"></content>
```

**Comment by yyx990803** [comment]
Fixed by merging #787.
